weave/trace/view_utils.py

```diff
@@ -0,0 +1,133 @@
+"""Utilities for attaching Content-backed views to trace calls."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from weave.trace.call import Call
+from weave.trace.serialization.serialize import to_json
+from weave.trace.weave_client import WeaveClient
+from weave.type_wrappers.Content.content import Content
+
+
+def resolve_view_content(
+    content: Content | str,
+    *,
+    extension: str | None = None,
+    mimetype: str | None = None,
+    metadata: dict[str, Any] | None = None,
+    encoding: str = "utf-8",
+) -> Content:
+    """Return a ``weave.Content`` for the provided content input.
+
+    Args:
+        content: Either a preconstructed ``weave.Content`` instance or a text
+            string to encode as content.
+        extension: Optional file extension (e.g. ``"md"`` or ``".html"``)
+            used when the ``content`` argument is a string.
+        mimetype: Explicit MIME type override for string content.
+        metadata: Optional metadata to attach when constructing content from a
+            string.
+        encoding: Text encoding to use when converting string content.
+
+    Returns:
+        A ``weave.Content`` instance that can be serialized into the call
+        summary.
+
+    Raises:
+        TypeError: If ``content`` is neither a string nor a ``weave.Content``.
+
+    Examples:
+        >>> resolve_view_content("# Title", extension="md").mimetype
+        'text/markdown'
+    """
+    if isinstance(content, Content):
+        return content
+
+    if isinstance(content, str):
```

Comment:
Mimetype can't be easily resolved for text type contents so this is going to typically show up as text/plain if the default args are used just a heads up.